### PR TITLE
Add DRIZPARS keyword to drizzle header

### PR DESCRIPTION
**Pull request by stsci-hack**
These changes add a new keyword, _DRIZPARS_, to the Primary header of the final drizzle product which provides the filename of the trailer file (or logfile) that contains the processing comments generated when creating the drizzle product.  

In addition, the `adrizzle.py` module was cleaned up to meet the latest Flake8 standards.

### CODE REVIEW

**Comment by stsci-hack** [comment]
As of 8ff85fb, the changes have been verified by processing:
- SVM processing: ib2t06
- runastrodriz (pipeline) processing:  j9io03040
- stand-alone drizzling: j9io03040 

This verified that AstroDrizzle will put the correct value of the trailer file in the trailer file itself, as well as the DRIZPARS keyword.  Both 'runastrodriz' and SVM processing ('hapsequencer') set up a temp file then build up the final trailer file from all the temp files, so those scripts update the DRIZPARS keyword on their own using the filenames they define for the trailer files.

File: drizzlepac/haputils/product.py [hunk 1]
```diff
@@ -161,6 +161,21 @@ def __init__(self, prop_id, obset_id, instrument, detector, filename, filetype,
...
+
+        # Update all filter drizzle products that were actually written out
+        for fdp in self.fdp_list:
+            if os.path.exists(fdp.drizzle_filename):
+                fits.setval(fdp.drizzle_filename, 'DRIZPARS', value=fdp.trl_filename)
+
+        # Update all exposure drizzle products that were actually written out
+        for edp in self.edp_list:
+            if os.path.exists(edp.drizzle_filename):
+                fits.setval(edp.drizzle_filename, 'DRIZPARS', value=edp.trl_filename)
```
Comment (mdlpstsci) [inline drizzlepac/haputils/product.py:177]: Curious why there is a check for the individual exposure product object, but not for the filter product?

File: drizzlepac/util.py [hunk 2]
```diff
@@ -376,14 +381,14 @@ def _ptime():
...
     else:
         # Basic time stamp which only includes integer seconds
         # Format time values for keywords IRAF-TLM, and DATE
         _ltime = time.localtime(ftime)
-        tlm_str = time.strftime('%H:%M:%S (%d/%m/%Y)',_ltime)
-        #date_str = time.strftime('%Y-%m-%dT%H:%M:%S',_ltime)
-    return tlm_str,ftime
+        tlm_str = time.strftime('%H:%M:%S (%d/%m/%Y)', _ltime)
+        # date_str = time.strftime('%Y-%m-%dT%H:%M:%S',_ltime)
+    return tlm_str, ftime
```
Comment (mdlpstsci) [inline drizzlepac/util.py:391]: Insert space before _ltime at line 391 although this line of code seems to be deprecated and could be deleted.

File: drizzlepac/util.py [hunk 3]
```diff
@@ -362,8 +367,8 @@ def reportTimes(self):
         print("", flush=True)
 
         # Compute overall runtime of entire program, including overhead
-        #total = self.end[1] - self.start[1]
-        #print '   %20s          %0.4f sec.'%('Total Runtime',total)
+        # total = self.end[1] - self.start[1]
+        # print '   %20s          %0.4f sec.'%('Total Runtime',total)
```
Comment (mdlpstsci) [inline drizzlepac/util.py:372]: Should lines 370-371 be deleted?

File: drizzlepac/util.py [hunk 4]
```diff
@@ -269,24 +274,24 @@ def output(msg):
...
+            except Exception:
                 vstr += "No version defined.  "
             if svn:
                 git = True
-                #try:
+                # try:
                 #    vstr += "\n    SVN version -> "+package.__svn_version__.rstrip()
-                #except:
+                # except:
                 #    vstr += " "
```
Comment (mdlpstsci) [inline drizzlepac/util.py:287]: Does the SVN block need to exist?  If so, is just git = True sufficient?

File: drizzlepac/util.py [hunk 5]
```diff
@@ -719,31 +724,31 @@ def getDefaultConfigObj(taskname,configObj,input_dict={},loadOnly=True):
...
-        validateUserPars(configObj,input_dict)
+        validateUserPars(configObj, input_dict)
 
         # If everything looks good, merge user inputs with configObj and continue
         cfgpars.mergeConfigObj(configObj, input_dict)
         # Update the input .cfg file with the updated parameter values
-        #configObj.filename = os.path.join(cfgpars.getAppDir(),os.path.basename(configObj.filename))
-        #configObj.write()
+        # configObj.filename = os.path.join(cfgpars.getAppDir(),os.path.basename(configObj.filename))
+        # configObj.write()
```
Comment (mdlpstsci) [inline drizzlepac/util.py:737]: Are lines 736 and 737 deprecated?

File: drizzlepac/adrizzle.py [hunk 6]
```diff
@@ -856,26 +864,26 @@ def run_driz_chip(img,chip,output_wcs,outwcs,template,paramDict,single,
...
             # applying the static mask
-            mergeDQarray(crMaskName,dqarr)
+            mergeDQarray(crMaskName, dqarr)
 
             if dqarr.sum() == 0:
                 log.warning('WARNING: All pixels masked out when applying '
                             'cosmic ray mask to %s' % _expname)
-        updateInputDQArray(chip.dqfile,chip.dq_extn,chip._chip,
+        updateInputDQArray(chip.dqfile ,chip.dq_extn, chip._chip,
                            crMaskName, paramDict['crbit'])
```
Comment (mdlpstsci) [inline drizzlepac/adrizzle.py:884]: Line 883 (eyes are gaga): space added in wrong place - remove after chip.dqfile and put after comma.

**Review by mdlpstsci — DISMISSED** [review]
The important changes to address the actual issue are fine.  The remainder of the changes are associated with Pep 8 or Flake 8 which are also fine.  There are a few comments/questions on what appears to be deprecated code.  This code should be cleaned up eventually, but the PR can go now as necessary.

File: drizzlepac/haputils/product.py (hunk 1 above)
Comment (stsci-hack) [reply drizzlepac/haputils/product.py:177]: Lines 170-172 adds the keyword to the filter products with Line 171 checking to see that it exists in the first place.  This seems to address your question.

File: drizzlepac/adrizzle.py (hunk 6 above)
Comment (stsci-hack) [reply drizzlepac/adrizzle.py:884]: Corrected in b11537b4

File: drizzlepac/util.py (hunk 5 above)
Comment (stsci-hack) [reply drizzlepac/util.py]: Deleted in b11537b4

File: drizzlepac/util.py (hunk 4 above)
Comment (stsci-hack) [reply drizzlepac/util.py]: This section deleted in b11537b4

File: drizzlepac/util.py (hunk 3 above)
Comment (stsci-hack) [reply drizzlepac/util.py]: Lines were deleted in b11537b4

File: drizzlepac/util.py (hunk 2 above)
Comment (stsci-hack) [reply drizzlepac/util.py:380]: Line 390 was deleted in b11537b4 (just to keep things simpler...)

**Review by mdlpstsci — APPROVED** [review]

**Pull request merged**